packages/web/src/components/change-password/ChangePasswordModal.tsx

```diff
@@ -39,34 +46,44 @@ const messages = {
 }
 
 export const ConfirmCredentialsPage = () => {
+  const [emailField] = useHarmonyField('email')
+  const [passwordField] = useHarmonyField('password')
   return (
     <Flex direction='column' gap='xl'>
       <Text variant='body'>{messages.confirmPasswordHelp}</Text>
-      <HarmonyTextField
-        name='email'
-        autoComplete='email'
+      <TextInput
+        {...emailField}
         label={messages.email}
+        autoComplete='email'
         autoFocus
       />
-      <HarmonyPasswordField name='password' label={messages.currentPassword} />
+      <PasswordInput
```

Comment:
I like the hook idea a lot! I think though we should still have ready to go field components though so users aren't having to useHarmonyField and then also pass that into an input, what do you think?

Reply:
I personally would much prefer passing in the props to an input than a wrapper component, but I think this is a style/taste thing. Either way, the hook can be used to make all 4 variants of those Field components to keep them all in sync with the same logic (mobile/web, passwordfield/textfield) if we decide to do wrappers

Reply:
Actually come around on this a bit and I think requiring a hook is bad, but I think a generic `Field` component would be ideal, like the [original Formik one](https://formik.org/docs/api/field), rather than unique Field types. So like:

```
<Field asChild name="email"><TextInput /></Field>
```

```
<Field component={TextInput} name="email" />
```

We'd have to standardize on props for inputs though.

It seems the only way we deviate from standard inputs is error styling (`error`) and helper text (`helperText`), which we could maybe build into our own field component prop standard?

Alternatively yeah individual field components for each thing doesn't feel _that_ bad... It just confuses me when searching for what component I want to use

Reply:
Yeah the main issue I found with the general field prop is to make sure the props map correctly which is why I kinda viewed the current approach as the best next thing, a bit of duplication which we could share with a hook

Reply:
My mental model is "component-name+field" for formik connected components, with a caveat that input is replaced with field if it exists

Reply:
Yeah now that I know it's not so bad, but it does feel almost redundant with how little unique logic is in the Field components

also super minor but this makes me think for a few extra milliseconds 😂 
![image](https://github.com/AudiusProject/audius-protocol/assets/3690498/889ac716-279a-4043-bb10-535d133a0ab9)